shared/settings/advanced/container.js

```diff
@@ -15,13 +16,15 @@ import {connect, lifecycle} from '../../util/container'
 
 type OwnProps = {||}
 const mapStateToProps = state => ({
+  hasRandomPW: state.settings.passphrase.randomPW,
```

Comment:
I think this should be `!!state.settings.passphrase.randomPW`, flow is letting it pass because the `lifecycle` below is losing the component's type.